### PR TITLE
feat(comp): add new Component Switch

### LINKED ISSUE #1653: Component - Switch
## Description

Create a toggle switch component to choose between two states.
- [Switch Design](https://www.figma.com/file/DFKyTGHAoDxOsUBPszLLxP/%F0%9F%8F%B5%EF%B8%8FMarigold?node-id=2458%3A2252)
- [react-aria useSwitch](https://react-spectrum.adobe.com/react-aria/useSwitch.html)

## Context

Have one more component we need in future

## Use Cases 

First usage could be in the new rx Checkout or in marigold docs to switch between dark and light theme

## Consequences

Easier to use a Switch functionality in an application which uses marigold

### PR DESCRIPTION
**Pull request by ti10le**
closes #1653

### CODE REVIEW

File: packages/components/src/Switch/Switch.test.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import React from 'react';
```
Comment (sebald) [inline packages/components/src/Switch/Switch.test.tsx:1]: Add a test that shows the controlled state of the component.

File: packages/components/src/Switch/Switch.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  disabled?: boolean;
+} & AriaSwitchProps &
+  ToggleProps &
+  ComponentProps<'input'>;
+
+// Component
+// ---------------
+export const Switch: React.FC<SwitchProps> = ({
+  variant = '',
+  labelVariant = 'above',
```
Comment (sebald) [inline packages/components/src/Switch/Switch.tsx:37]: Can we use a child like in `<Checkbox>`?

File: packages/components/src/Switch/Switch.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        <input
+          {...inputProps}
+          {...focusProps}
+          disabled={disabled}
+          ref={ref as LegacyRef<HTMLInputElement>}
+        />
+      </VisuallyHidden>
+      <Box
+        as="svg"
+        __baseCSS={{
```
Comment (sebald) [inline packages/components/src/Switch/Switch.tsx:63]: add `userSelect: 'none'` so text doesn't selected on accident

Comment (sebald) [reply packages/components/src/Switch/Switch.tsx:63]: image

**Review by sebald — CHANGES_REQUESTED** [review]

File: packages/components/src/Switch/Switch.tsx (hunk 2 above)
Comment (ti10le) [reply packages/components/src/Switch/Switch.tsx:37]: Yes I do it with props.children but I can also change to children

File: packages/components/src/Switch/Switch.tsx (hunk 3 above)
Comment (sebald) [reply packages/components/src/Switch/Switch.tsx:63]: No in the baseCSS prop :)

Comment (ti10le) [reply packages/components/src/Switch/Switch.tsx:63]: but this has no effect

Comment (ti10le) [reply packages/components/src/Switch/Switch.tsx:63]: I've tried it like this but no effect: https://css-tricks.com/almanac/properties/u/user-select/

**Comment by ti10le** [comment]
@sebald its ready now :)

**Review by sebald — APPROVED** [review]

**Pull request merged**